tests/framework/web/helpers/CHtmlTest.php

```diff
@@ -328,6 +352,42 @@ public function testScript($text, $assertion)
 	{
 		$this->assertEquals($assertion, CHtml::script($text));
 	}
+	
+	public static function providerScriptWithoutCdata()
+	{
+		return array(
+			array('var a = 10;', "<script type=\"text/javascript\">\nvar a = 10;\n</script>"),
+			array("\t(function() { var x = 100; })();\n\tvar y = 200;",
+				"<script type=\"text/javascript\">\n\t(function() { var x = 100; })();\n\tvar y = 200;\n</script>"),
+		);
+	}
+
+	/**
+	 * @dataProvider providerScriptWithoutCdata
+	 * @backupStaticAttributes enabled
+	 *
+	 * @param string $text
+	 * @param string $assertion
+	 */
+	public function testScriptWithoutCdata($text, $assertion)
+	{
+		CHtml::$cdataScriptAndStyleContents=false;
+		$this->assertEquals($assertion, CHtml::script($text));
+	}
+	
+	/**
+	 * @dataProvider providerScript
+	 * @backupStaticAttributes enabled
+	 *
+	 * @param string $text
+	 * @param string $assertion
+	 */
+	public function testScriptHtml5($text, $assertion)
+	{
+		CHtml::$setScriptType=false;
```

Comment:
Shouldn't we reset this flag after end of the test?

Reply:
That is what the `@backupStaticAttributes enabled` annotation should be taking care of. (https://phpunit.readthedocs.io/en/9.5/annotations.html#backupstaticattributes). But note that this is a very broad approach - it backups and resets any static class properties.

Is it perhaps better to explicitly reset the required static attributes in_setUp() ?

Reply:
PHPUnit always has multiple ways to achieve the same thing. :) I try to adhere to the existing testing style as much as possible and not introduce new concepts, but this is used in a few other places as well.